src/app/modules/timesheet/components/timesheet-edit/timesheet-edit.component.ts

```diff
@@ -99,12 +113,14 @@ export class TimesheetEditComponent implements OnInit {
   }
 
   updateMailtoLink(): void {
-    this.reviewMail = new ReviewMail(
-      this.timesheetService.timesheet,
-      this.calendarService.getWorkedTime(this.timesheetService.timesheet),
-      this.timesheetService.getReviewToken(),
-      this.originUrl + '/timesheet/review/'
-    );
+    this.timesheetService.shortenUrl(this.originUrl + '/timesheet/review/' + this.timesheetService.getReviewToken())
```

Comment:
You can't get the `reviewShortUrl` instead of rebuild it ? 
You will need to call the `setShortUrls` on the constructor to be sure to have all time an URL set.